spock-core/src/main/java/org/spockframework/runtime/StackTraceFilter.java

```diff
@@ -58,6 +60,14 @@ public StackTraceFilter(IMethodNameMapper mapper) {
 
   @Override
   public void filter(Throwable throwable) {
+    filter(throwable, new HashSet<>());
```

Comment:
If we add this, then we should use the same method that the `java.lang.Throwable#lockedPrintStackTrace` uses
```suggestion
    filter(throwable, Collections.newSetFromMap(new IdentityHashMap<>()));
```